dockers/docker-sonic-mgmt-framework/Dockerfile.j2

```diff
@@ -46,11 +32,11 @@ COPY ["start.sh", "rest-server.sh", "/usr/bin/"]
 COPY ["mgmt_vars.j2", "/usr/share/sonic/templates/"]
 COPY ["supervisord.conf", "/etc/supervisor/conf.d/"]
 
-# TODO: Remove this line once we no longer need Python 2
-RUN apt-get purge -y python-dev
-
 RUN apt-get remove -y g++ python3-dev
 RUN apt-get clean -y; apt-get autoclean -y; apt-get autoremove -y
 RUN rm -rf /debs
 
+RUN ln -sf /host_etc/passwd /etc/passwd	
```

Comment:
Could you please explain the need for this mapping?

Reply:
As of now, the rest server performs basic auth.
 https://github.com/Azure/sonic-mgmt-framework/blob/master/rest/server/pamAuth.go#L117
This will be replaced with JWT token based authentication.

Reply:
I believe d-bus mechanism was the *discussed* solution.

Reply:
This PR is limited to start enhanced sonic-CLI and python3 support. I am not aware of the d-bus based solution, but I will ask someone who is familiar with it to have a separate PR to upstream the d-bus based solution.

Reply:
@renukamanavalan 
According to the local discussion, it is found that we still need to mount /etc/passwd and /etc/group for rest server to perform group lookups. Once “AAA Improvements” ([HAM](https://github.com/Azure/sonic-buildimage/pull/5553)) is merged to SONiC, it can be obviated since the NSS module will take care of the user/group lookups and there is no need to mount or link the /etc/passwd and /etc/group files anymore. Since the AAA improvements have not been merged, we still need those files accessible in the mgmt-framework. 

As far as security concerned, it should be ok because we are not exposing anything other than what a SONiC user already has access to. Please share your concern if you have.

Reply:
I would like to get a review from @lguohan

Reply:
Just to clarify more specifically, see:
https://github.com/martin-belanger/SONiC/blob/aaa_improvements/doc/aaa/AAA%20Improvements/AAA%20Improvements.md#hamd-for-containers

Specifically, with the AAA improvements, the NSS module will take care of the user/group getpwnam lookups and you don’t need to mount or link the /etc/passwd and /etc/group files anymore.

Reply:
can we then just add some comments saying such will be removed when AAA improvements has been merged? do we have AAA improvement pr?

Reply:
@lguohan the comment explicitly added as requested, 
The PR for AAA improvements is https://github.com/Azure/sonic-buildimage/pull/5553